client/src/pages/board/KanbanBoard.jsx

```diff
@@ -0,0 +1,297 @@
+import React, { useState, useEffect, memo } from 'react';
+import { DragDropContext, Droppable } from 'react-beautiful-dnd';
+import { makeStyles } from '@material-ui/core/styles';
+import { CardActions, Grid } from '@material-ui/core';
+import Column from './components/Column';
+
+import {getBoard, saveBoard} from '../../api/Board';
+
+import AddColumnSidebar from './components/dashboardUI/AddColumnSidebar';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+  },
+  column: {
+    padding: theme.spacing(1),
+    paddingTop: theme.spacing(2),
+    margin: theme.spacing(1),
+    minHeight: '30vh',
+    backgroundColor: '#F4F6FF',
+    borderRadius: 4,
+  },
+  drag: {
+    width: 275, 
+    minHeight: '20vh',
+    padding: theme.spacing(1),
+    borderRadius: 4,
+    backgroundColor: '#F4F6FF',
+  },
+  card: {
+    userSelect: 'none',
+    minHeight: '50px',
+    padding: theme.spacing(2),
+    margin: theme.spacing(1),
+  },
+  button: {
+    color: 'black',
+    backgroundColor: '#F4F6FF',
+    '&:hover': {
+      color: 'white',
+      backgroundColor: '#759CFC',
+    },
+    margin: theme.spacing(1),
+    padding: theme.spacing(1),
+    textTransform: 'none',
+    alignContent: 'center',
+  },
+  addColumnContainer: {
+    width: '30vh',
+    overflow: 'hidden',
+    '&#leftNav': {
+      '& .addColumnContent': {
+        minHeight: '500px',
+        '& a': {
+          width: '2%',
+          padding: '0 30px 0 0',
+          left: '-30px',
+          minHeight: '500px',
+          position: 'fixed',
+          textDecoration: 'none',
+          height: 'inherit',
+          display: 'flex',
+          alignItems: 'center',
+          '& button': {
+            marginRight: '10px',
+            backgroundColor: 'lightgrey',
+            padding: '0px',
+            color: 'grey',
+          },
+          '&:hover': {
+            padding: '0 0 0 20px',
+            left: 0,
+            position: 'relative',
+            width: 'fit-content',
+            backgroundColor: 'lightgrey',
+            transition: '0.05s',
```

Comment:
Hey @nicktran6997 ! Discussed this with @swathivottigunta and I'll be letting her handle this on her branch since it's her code that I had to refactor in my own components to get it working locally.